projects/ngx-monaco-tree/src/lib/monaco-tree-file/monaco-tree-file.component.ts

```diff
@@ -1,67 +1,152 @@
 import { extensions } from '../../utils/extension-icon';
 import { files } from '../../utils/file-icon';
 import { folders } from '../../utils/folder-icon';
 import { MonacoTreeElement } from '../ngx-monaco-tree.type';
 import {
-	ContextMenuElementSeparator,
-	ContextMenuElementText
+  ContextMenuElementSeparator,
+  ContextMenuElementText
 } from "../monaco-tree-context-menu/monaco-tree-context-menu.type";
-import {ContextMenuAction, DragAndDropEvent} from "./monaco-tree-file.type";
-import {MonacoTreeContextMenuComponent} from "../monaco-tree-context-menu/monaco-tree-context-menu.component";
-import {NgForOf, NgIf, NgStyle} from "@angular/common";
-import {DragDropModule, CdkDragDrop, CdkDrag, CdkDropList} from '@angular/cdk/drag-drop';
+import { ContextMenuAction, DragAndDropEvent } from "./monaco-tree-file.type";
+import { MonacoTreeContextMenuComponent } from "../monaco-tree-context-menu/monaco-tree-context-menu.component";
+import { NgClass, NgStyle } from "@angular/common";
+import { DragDropModule, CdkDragDrop, CdkDrag, CdkDropList } from '@angular/cdk/drag-drop';
 
 function getAbsolutePosition(element: any) {
-	const r = { x: element.offsetLeft, y: element.offsetTop };
-	if (element.offsetParent) {
-		const tmp = getAbsolutePosition(element.offsetParent);
-		r.x += tmp.x;
-		r.y += tmp.y;
-	}
-	return r;
+  const r = { x: element.offsetLeft, y: element.offsetTop };
+  if (element.offsetParent) {
+    const tmp = getAbsolutePosition(element.offsetParent);
+    r.x += tmp.x;
+    r.y += tmp.y;
+  }
+  return r;
 };
 
 @Component({
   selector: 'monaco-tree-file',
-  standalone: true,
-  imports: [NgIf, NgForOf, MonacoTreeContextMenuComponent, DragDropModule, NgStyle, CdkDrag, CdkDrag, CdkDropList],
+  imports: [
+    MonacoTreeContextMenuComponent,
+    DragDropModule,
+    NgClass,
+    NgStyle,
+    CdkDrag,
+    CdkDrag,
+    CdkDropList
+  ],
   templateUrl: './monaco-tree-file.component.html',
   styleUrls: ['./monaco-tree-file.component.scss']
 })
 export class MonacoTreeFileComponent implements OnChanges {
-	@Input() name = '';
-  @Input() path = '';
-  @Input() color?: string|null|undefined = '';
-	@Input() content: MonacoTreeElement[]|undefined|null = undefined;
-	@Input() depth = 0;
-	@Input() theme: 'vs-dark'|'vs-light' = 'vs-dark';
-	@Input() hide = false;
-  @Input() current: string|null = null;
-
-	@Output() clickFile = new EventEmitter<string>();
-	@Output() contextMenuClick = new EventEmitter<ContextMenuAction>();
-  @Output() dragDropFile = new EventEmitter<DragAndDropEvent>();
-
-  private children = viewChildren(MonacoTreeFileComponent);
-
-	open = false;
-	position: [number, number]|undefined = undefined;
-
+  private readonly eRef = inject(ElementRef);
+  private readonly children = viewChildren(MonacoTreeFileComponent);
+
+  readonly name = input.required<string>();
+  readonly path = input('');
+  readonly color = input<string>();
+  readonly content = input.required<MonacoTreeElement[] | undefined>();
```

Comment:
this looks strange to have an `input.required` accepting `undefined`, is that a mistake?